baseapp/abci_test.go

```diff
@@ -44,6 +42,21 @@ func TestABCI_Info(t *testing.T) {
 	require.Equal(t, suite.baseApp.AppVersion(), res.AppVersion)
 }
 
+func TestABCI_First_block_Height(t *testing.T) {
+	suite := NewBaseAppSuite(t, baseapp.SetChainID("test-chain-id"))
+	app := suite.baseApp
+
+	app.InitChain(abci.RequestInitChain{
+		ChainId:         "test-chain-id",
+		ConsensusParams: &cmtproto.ConsensusParams{Block: &cmtproto.BlockParams{MaxGas: 5000000}},
+		InitialHeight:   1,
+	})
+	_ = app.Commit()
+
+	ctx := app.GetContextForCheckTx(nil)
+	require.Equal(t, int64(1), ctx.BlockHeight())
```

Comment:
w/o this fix, this value is actually 0 which would cause `CheckTx` to fail for txs prior to the 1st block.